### PR TITLE
feat: Add station_id field to measurements storage and publishing

### PR DESCRIPTION
**Pull request by PatrickNercessian**
DRAFT, DO NOT MERGE

Links:
filecoin-station/roadmap#96

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -67,6 +67,7 @@ const createMeasurement = async (req, res, client, getCurrentRound) => {
   validate(measurement, 'indexerResult', { type: 'string', required: false })
   validate(measurement, 'minerId', { type: 'string', required: false })
   validate(measurement, 'providerId', { type: 'string', required: false })
+  validate(measurement, 'stationId', { type: 'string', required: false }) // TODO eventually should be required, optional now for backwards compatibility
```
Comment (bajtos) [inline index.js:70]: 👌🏻

File: index.js [hunk 2]
```diff
@@ -91,10 +92,11 @@ const createMeasurement = async (req, res, client, getCurrentRound) => {
         indexer_result,
         miner_id,
         provider_id,
-        completed_at_round
+        completed_at_round,
+        station_id
```
Comment (bajtos) [inline index.js:96]: Let's avoid code churn. (It's a pity that standard.js does not allow trailing commas here 😢 )

```diff
   provider_id,
+  station_id,
   completed_at_round
```

Personally, I would put this new item right after `participant_address`, but that's a nitpick - feel free to ignore it.

File: index.js [hunk 3]
```diff
@@ -117,7 +119,8 @@ const createMeasurement = async (req, res, client, getCurrentRound) => {
     measurement.indexerResult,
     measurement.minerId,
     measurement.providerId,
-    sparkRoundNumber
+    sparkRoundNumber,
+    measurement.stationId
```
Comment (bajtos) [inline index.js:123]: You will need to update this depending on how you resolve my comment above.

File: test/test.js [hunk 4]
```diff
@@ -32,7 +32,8 @@ const VALID_MEASUREMENT = {
   carChecksum: 'somehash',
   minerId: 'f02abc',
   providerId: 'provider-pubkey',
-  indexerResult: 'OK'
+  indexerResult: 'OK',
+  stationId: '123e4567-e89b-12d3-a456-426614174000'
```
Comment (bajtos) [inline test/test.js:36]: IIRC, we are not going to use GUIDs for station IDs. Let's use a different value in the test data then, e.g. `some-station-id`.

**Review by bajtos — COMMENTED** [review]
Looks mostly good! 👏🏻

File: test/test.js (hunk 4 above)
Comment (PatrickNercessian) [reply test/test.js:36]: Ah yes, remnant of when I was building this with UUID in mind, good catch!

**Review by bajtos — APPROVED** [review]
LGTM 👍🏻 

Feel free to open another pull request to implement the validation rule accepting `stationId` values of a given string length only.

**Comment by PatrickNercessian** [comment]
> LGTM 👍🏻
> 
> Feel free to open another pull request to implement the validation rule accepting `stationId` values of a given string length only.

**Comment by bajtos** [comment]
@PatrickNercessian What's the status of this pull request? Is there anything left to improve or is it good to be landed?

**Comment by PatrickNercessian** [comment]
@bajtos I was going to do new PR for stationId constraints as you mentioned, but noticed there were some other code churn updates that could be reduced as we did in other PRs, so I thought I'd combine both.

Let me know your thoughts.

File: index.js [hunk 5]
```diff
@@ -67,7 +67,10 @@ const createMeasurement = async (req, res, client, getCurrentRound) => {
   validate(measurement, 'indexerResult', { type: 'string', required: false })
   validate(measurement, 'minerId', { type: 'string', required: false })
   validate(measurement, 'providerId', { type: 'string', required: false })
-  validate(measurement, 'stationId', { type: 'string', required: false }) // TODO eventually should be required, optional now for backwards compatibility
+  if ('stationId' in measurement) {
+    validate(measurement, 'stationId', { type: 'string', required: false }) // TODO eventually should be required, optional now for backwards compatibility
+    assert(measurement.stationId.length === 64, 400, 'Invalid Station ID')
```
Comment (bajtos) [inline index.js:72]: How strict do we want to make this assertion? Should we also check that string contains only hex-encoding chars? (E.g. `/^[0-9A-Z]{64}$/`).

**Review by bajtos — APPROVED** [review]

File: index.js (hunk 5 above)
Comment (juliangruber) [reply index.js:72]: Good idea! I think [0-9A-F]

**Review by juliangruber — APPROVED** [review]

**Pull request merged**